async_processor/types.py

```diff
@@ -216,6 +221,20 @@ def to_input(self) -> Input:
         return SQSInput(self)
 
 
+class AMQPInputConfig(InputConfig):
+    type: constr(regex=r"^amqp$") = "amqp"
+
+    hostname: str
+    port: Optional[int] = 5672
+    auth: AMQPUserPasswordAuth
```

Comment:
Should authentication always be mandatory?
Also, is this the same authentication method that https://console.cloud.google.com/marketplace/product/codes-public/cloudamqp.endpoints.codes-public.cloud.goog uses?